### PR TITLE
Getting started with TypeScript example

### PR DESCRIPTION
**Pull request by anilkk**
#### Summary
Tutorial for getting started with TypeScript SDK.

#### Todo

- Tests
  - [ ] Unit
  - [ ] Integration
  - [ ] Acceptance
- [ ] Documentation
- [ ] `Type` label for the PR

### CODE REVIEW

File: docs/sdk/api/typescriptSdk.md [hunk 1]
```diff
@@ -25,6 +25,126 @@ npm install --save @commercetools/ml-sdk
...
+Replace the values with your client id and client secret that you copied earlier.
+
+Re-open `project.js` and add the following code:
+```js
+const { 
+    ADMIN_CLIENT_ID,
+    ADMIN_CLIENT_SECRET,
+} = process.env;
+
+const projectKey = '<your_project_key>'
```
Comment (stmeissner) [inline docs/sdk/api/typescriptSdk.md:90]: Could we read this from the `.env` file as well? Then this part does not have to be edited by the reader.

File: docs/sdk/api/typescriptSdk.md [hunk 2]
```diff
@@ -25,6 +25,126 @@ npm install --save @commercetools/ml-sdk
...
+
+// Create a httpMiddleware for the your project AUTH URL
+const authMiddleware = createAuthMiddlewareForClientCredentialsFlow({
+    host: '<your_auth_url>',
```
Comment (stmeissner) [inline docs/sdk/api/typescriptSdk.md:94]: Could we read this from the `.env` file as well? Then this part does not have to be edited by the reader.

File: docs/sdk/api/typescriptSdk.md [hunk 3]
```diff
@@ -25,6 +25,126 @@ npm install --save @commercetools/ml-sdk
...
+    projectKey,
+    credentials: {
+        clientId: ADMIN_CLIENT_ID,
+        clientSecret: ADMIN_CLIENT_SECRET,
+    },
+    scopes: ['<your_client_scope>'],
```
Comment (stmeissner) [inline docs/sdk/api/typescriptSdk.md:100]: Could we read this from the `.env` file as well? Then this part does not have to be edited by the reader.

File: docs/sdk/api/typescriptSdk.md [hunk 4]
```diff
@@ -25,6 +25,126 @@ npm install --save @commercetools/ml-sdk
...
+    fetch,
+})
+
+// Create a httpMiddleware for the your project API URL
+const httpMiddleware = createHttpMiddleware({
+    host: '<your_api_url>',
```
Comment (stmeissner) [inline docs/sdk/api/typescriptSdk.md:106]: Could we read this from the `.env` file as well? Then this part does not have to be edited by the reader.

File: docs/sdk/api/typescriptSdk.md [hunk 5]
```diff
@@ -25,6 +25,115 @@ npm install --save @commercetools/ml-sdk
...
+
+### Create a API client
+[Create API client](https://docs.commercetools.com/tutorials/getting-started#creating-an-api-client) from Merchant Center. If you do not have account [follow the steps to create a free trial account](https://docs.commercetools.com/tutorials/getting-started#first-steps). 
+In this guide we’ll be calling a method of commercetools API using TypeScript SDK to get the settings of a commercetools project. The commercetools API is the foundation of the commercetools Platform, and almost every commercetools client app uses it. Aside from getting project information, the commercetools API allows clients to call methods that can be used for everything from creating products to updating an order’s status. Before we can call any methods, we need to configure our new app to obtain an access token.
+
+### Set up your local project
+If you don’t already have a project, let’s create a new one. In an empty directory, you can initialize a new project using the following command:
```
Comment (stmeissner) [inline docs/sdk/api/typescriptSdk.md:36]: Can we say "Node.js project" here? I just want to be clear it cannot be confused with the commercetools project that should be existing at this step for sure.

File: docs/sdk/api/typescriptSdk.md [hunk 6]
```diff
@@ -25,6 +25,115 @@ npm install --save @commercetools/ml-sdk
...
+})();
+```
+Then client will **execute** get project information request from `apiRoot` using **TypeScript SDK**.
+Run the program. The output should look like the following if the request is successful:
+```
+$ node project.js
+Getting started with commercetools Typescript SDK
+Project information ---> {
+  body: {
+    <projectData>
```
Comment (stmeissner) [inline docs/sdk/api/typescriptSdk.md:131]:
```suggestion
    key: '<project_key>',
    name: '<project_name>',
    countries: [ ... ],
    currencies: [ ... ],
    languages: [ ... ],
    createdAt: '...',
    ...
```

**Review by stmeissner — APPROVED** [review]
LGTM 👍 +1

File: docs/sdk/api/typescriptSdk.md (hunk 5 above)
Comment (anilkk) [reply docs/sdk/api/typescriptSdk.md:36]: Yes

File: docs/sdk/api/typescriptSdk.md [hunk 7]
```diff
@@ -29,6 +29,120 @@ npm install --save @commercetools/ml-sdk
 </script>
 ```
 
+## Getting started
+This tutorial will show you how to use the middlewares in this **[commercetools TypeScript SDK](https://github.com/commercetools/commercetools-sdk-typescript/)** to get a simple commercetools JavaScript app running.
+
+### Create a API client
+[Create API client](https://docs.commercetools.com/tutorials/getting-started#creating-an-api-client) from Merchant Center. If you do not have account [follow the steps to create a free trial account](https://docs.commercetools.com/tutorials/getting-started#first-steps).
```
Comment (industrian) [inline docs/sdk/api/typescriptSdk.md:36]:
```suggestion
[Create an API client](https://docs.commercetools.com/tutorials/getting-started#creating-an-api-client) from the Merchant Center.

If you do not have account [follow these steps to create a free trial account](https://docs.commercetools.com/tutorials/getting-started#first-steps). 

When you have set up your API Client, ensure that you save your **project_key**, **client_id**, **secret**, **scope**, **API URL**, and **Auth URL**. We recommend using the drop-down list to save your API Client details as an **Environment Variables (.env)** file.
```

File: docs/sdk/api/typescriptSdk.md [hunk 8]
```diff
@@ -29,6 +29,120 @@ npm install --save @commercetools/ml-sdk
 </script>
 ```
 
+## Getting started
+This tutorial will show you how to use the middlewares in this **[commercetools TypeScript SDK](https://github.com/commercetools/commercetools-sdk-typescript/)** to get a simple commercetools JavaScript app running.
+
+### Create a API client
+[Create API client](https://docs.commercetools.com/tutorials/getting-started#creating-an-api-client) from Merchant Center. If you do not have account [follow the steps to create a free trial account](https://docs.commercetools.com/tutorials/getting-started#first-steps). 
+In this guide we’ll be calling a method of commercetools API using TypeScript SDK to get the settings of a commercetools project. The commercetools API is the foundation of the commercetools Platform, and almost every commercetools client app uses it. Aside from getting project information, the commercetools API allows clients to call methods that can be used for everything from creating products to updating an order’s status. Before we can call any methods, we need to configure our new app to obtain an access token.
```
Comment (industrian) [inline docs/sdk/api/typescriptSdk.md:37]:
```suggestion

In this guide we call a method of the commercetools API using the TypeScript SDK to get the settings of a commercetools project.

The commercetools API is the foundation of the commercetools Platform, and almost every commercetools client app uses it.

Aside from retrieving project information, the commercetools API allows clients to call methods that can be used for everything from creating products to updating an order’s status.

Before we can call any methods, we need to configure a new app to obtain an access token.
```

**Review by industrian — CHANGES_REQUESTED** [review]
The code itself is fine, but I made some restructuring and changes.

File: docs/sdk/api/typescriptSdk.md [hunk 9]
```diff
@@ -29,6 +29,148 @@ npm install --save @commercetools/ml-sdk
...
+Aside from retrieving project information, the commercetools API allows clients to call methods that can be used for everything from creating products to updating an order’s status.
+
+Before we can call any methods, we need to configure a new app to obtain an access token.
+
+### Set up your local project
+1. Open a terminal window.
+2. Create a new directory for the project.
+3. Open the new directory and create a new project using the following command: ```npm init -y```
+4. The project will be created and a package.json file will be created in the directory.
+5. Enter the following command to open this folder in your code editor ```code .````
```
Comment (industrian) [inline docs/sdk/api/typescriptSdk.md:55]:
```suggestion
5. Enter the following command to open this folder in your code editor: ```code .```
```

File: docs/sdk/api/typescriptSdk.md [hunk 10]
```diff
@@ -29,6 +29,148 @@ npm install --save @commercetools/ml-sdk
...
+Before we can call any methods, we need to configure a new app to obtain an access token.
+
+### Set up your local project
+1. Open a terminal window.
+2. Create a new directory for the project.
+3. Open the new directory and create a new project using the following command: ```npm init -y```
+4. The project will be created and a package.json file will be created in the directory.
+5. Enter the following command to open this folder in your code editor: ```code .```
+
+```
```
Comment (industrian) [inline docs/sdk/api/typescriptSdk.md:57]:
```suggestion

```

File: docs/sdk/api/typescriptSdk.md [hunk 11]
```diff
@@ -29,6 +29,147 @@ npm install --save @commercetools/ml-sdk
...
+npm install @commercetools/sdk-client @commercetools/sdk-middleware-auth @commercetools/sdk-middleware-http @commercetools/typescript-sdk dotenv
+```
+
+This code creates a **client**, which uses `authMiddleware` and `httpMiddleware`. The `httpMiddleware` reads the `clientId` and `clientSecret` from the `.env` file.
+
+Run the program using the following command:
+```node project.js```
+You should see the following message in your output:
```
Comment (industrian) [inline docs/sdk/api/typescriptSdk.md:133]:
```suggestion

You should see the following message in your output:
```

File: docs/sdk/api/typescriptSdk.md [hunk 12]
```diff
@@ -29,6 +29,148 @@ npm install --save @commercetools/ml-sdk
...
+```
+
+This code creates a **client**, which uses `authMiddleware` and `httpMiddleware`. The `httpMiddleware` reads the `clientId` and `clientSecret` from the `.env` file.
+
+Run the program using the following command:
+```node project.js```
+
+You should see the following message in your output:
+```Getting started with commercetools Typescript SDK
+```
```
Comment (industrian) [inline docs/sdk/api/typescriptSdk.md:136]:
```suggestion

```

File: docs/sdk/api/typescriptSdk.md [hunk 13]
```diff
@@ -29,6 +29,147 @@ npm install --save @commercetools/ml-sdk
...
+
+This code creates a **client**, which uses `authMiddleware` and `httpMiddleware`. The `httpMiddleware` reads the `clientId` and `clientSecret` from the `.env` file.
+
+Run the program using the following command:
+```node project.js```
+
+You should see the following message in your output:
+```Getting started with commercetools Typescript SDK```
+If you have the same output as above, then you have set up everything correctly.
```
Comment (industrian) [inline docs/sdk/api/typescriptSdk.md:136]:
```suggestion

If you have the same output as above, then you have set up everything correctly.
```

Comment (industrian) [inline docs/sdk/api/typescriptSdk.md:137]:
```suggestion

If you have the same output as above, then you have set up everything correctly.
```

**Review by industrian — APPROVED** [review]

File: docs/sdk/api/typescriptSdk.md [hunk 14]
```diff
@@ -29,6 +29,149 @@ npm install --save @commercetools/ml-sdk
...
+npm install @commercetools/sdk-client @commercetools/sdk-middleware-auth @commercetools/sdk-middleware-http @commercetools/typescript-sdk dotenv
+```
+
+This code creates a **client**, which uses `authMiddleware` and `httpMiddleware`. The `httpMiddleware` reads the `clientId` and `clientSecret` from the `.env` file.
+
+Run the program using the following command:
+```node project.js```
+
+You should see the following message in your output:
+```Getting started with commercetools Typescript SDK```
```
Comment (industrian) [inline docs/sdk/api/typescriptSdk.md:135]:
```suggestion

```Getting started with commercetools Typescript SDK```
```

**Comment by ashishhk** [comment]
@ajimae can you please take a look at this PR and merge it if all looks good from technical correctness POV.
cc: @jenschude

**Review by ajimae — COMMENTED** [review]
Great work @anilkk, however, I think this example will suite more as the nodejs example rather than the TypeScript. This is because the way the [TypeScript](https://commercetools.github.io/nodejs/sdk/api/typescriptSdk.html#typescript-sdk-client-v2) is structured and used is a little different from this example.

So I will suggest you move this to a [nodejs example](https://commercetools.github.io/nodejs/sdk/api/sdkClient.html#usage-example) instead.

Cheers 🥂

**Comment by industrian** [comment]
@ashishhk Now that the new TypeScript SDK documentation is live, can this PR be closed?

**Comment by industrian** [comment]
Closing PR because:

1. The commercetools docs website has been updated with a getting started guide and example code for the TypeScript SDK.
2. This page has been updated to link to the commercetools docs website.

**Pull request closed without merging**